From e45a713ae920d8bcaa1da4b060fd70103a08d897 Mon Sep 17 00:00:00 2001
From: Benjamin Lin <benjamin-jw.lin@mediatek.com>
Date: Fri, 28 Jun 2024 11:11:47 +0800
Subject: [PATCH] hostapd: mtk: fix failure of BSS-wise QoS-map setting

wpa_driver_nl80211_data is PHY-wise data structure.
Therfore, using it here always sets QoS map of the first BSS of the PHY.
In order to correctly set QoS map of each BSS, BSS-wise data structure i802_bss is used.

Signed-off-by: Benjamin Lin <benjamin-jw.lin@mediatek.com>
---
 src/drivers/driver_nl80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index c1d31bc..29c4b9b 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -10496,7 +10496,7 @@ static int nl80211_set_qos_map(void *priv, const u8 *qos_map_set,
 	wpa_hexdump(MSG_DEBUG, "nl80211: Setting QoS Map",
 		    qos_map_set, qos_map_set_len);
 
-	if (!(msg = nl80211_drv_msg(drv, 0, NL80211_CMD_SET_QOS_MAP)) ||
+	if (!(msg = nl80211_bss_msg(bss, 0, NL80211_CMD_SET_QOS_MAP)) ||
 	    nla_put(msg, NL80211_ATTR_QOS_MAP, qos_map_set_len, qos_map_set)) {
 		nlmsg_free(msg);
 		return -ENOBUFS;
-- 
2.18.0

